src/controllers/credentials/index.ts

```diff
@@ -24,28 +27,38 @@ export class CredentialsController extends HTMLController {
     const credentials: Credential[] = await this.cloudagent.getCredentials()
     req.log.info('retrieved credentials from a cloudagent %j', credentials)
 
-    const combined = credentials
-      .map((cred: Credential) => {
-        if (cred.credentialAttributes) {
-          return {
-            ...cred,
-            updated_at: new Date(),
-            company_name: cred?.credentialAttributes[0].name === 'company_name' && cred.credentialAttributes[0].value,
-            company_number:
-              cred.credentialAttributes[1].name === 'company_number' && cred.credentialAttributes[1].value,
-          }
-        }
+    const formatted: Credentials = this.format(credentials)
+    for (let i = 0; i < formatted.length; i++) {
+      const [connection]: ConnectionRow[] = await this.db.get('connection', {
+        agent_connection_id: formatted[i].connectionId,
       })
-      .filter((cred) =>
-        search !== '' ? cred?.company_name.toString().toLowerCase().includes(search.toLowerCase()) : true
-      )
-
-    req.log.info('%j', combined, credentials)
-    if (search !== '') {
-      req.log.info('applying search filter... %j', combined) // do some mapping
+      formatted[i].connection = connection
     }
 
+    formatted.filter((cred) => {
+      if (!cred.attributes) return false
+      const companyName = cred.attributes.map(({ name }: { name: string }) => {
+        return name === 'company_name'
+      })
+      if (search !== '') return true
+      req.log.info('checking if %s includes %s', companyName, search)
+
+      return companyName.toString().toLowerCase().includes(search.toLowerCase())
```

Comment:
What is the intent of `companyName` here? You are mapping over the attributes and returning an equality check so `companyName` is a `boolean[]`. You're then `toString`ing it and filtering. Did you mean to do a `find` instead of a `map`?

Reply:
Find would work.
> What is the intent of companyName here?
to make sure that credential does have a company name that is a first column of a table. and do not render ones that do not have.

Reply:
This is not what this code is doing

Reply:
just updated to use find along with some additional tests